test/connection.js

```diff
@@ -0,0 +1,51 @@
+/* eslint-env node, mocha */
+require('./init.js');
+var assert = require('assert');
+var DataSource = require('loopback-datasource-juggler').DataSource;
+
+var config;
+
+before(function() {
+  config = global.getConfig();
+  console.log("config", config);
+});
+
+describe('testConnection', function() {
+  it('should pass with valid settings', function(done) {
+    var db = new DataSource(require('../'), config);
+    db.ping(function(err) {
+      assert(!err, 'Should connect without err.');
```

Comment:
@superkhau sorry I am a little confused what's the difference between `myFunction(done)` and 
```
myFunction(function(err) {
  if (err) return done(err);
})
```
If `myFunction` only has `error` returned?